rskj-core/src/main/java/co/rsk/pcc/altBN128/impls/AbstractAltBN128.java

```diff
@@ -16,20 +18,36 @@ public abstract class AbstractAltBN128 {
 
     protected byte[] output;
 
-    public static AbstractAltBN128 init() {
-        return init(Utils::isLinux, GoAltBN128::getLoadError);
+    /**
+     * Besides logging information about loaded library, this method has a side effect of enforcing native library loading,
+     * if it's not loaded yet. So that it allows to fail early, if, for instance, the loaded library is not compatible
+     * with the current environment.
+     */
+    public static void init() {
+        AbstractAltBN128 altBn128 = Objects.requireNonNull(create());
+
+        if (Utils.isLinux() && !(altBn128 instanceof GoAltBN128)) {
```

Comment:
Just a minor and picky thing, why have you "moved" logs here? I thing they belong to the place where the "selection" is made, this way you won't need this check and this `instanceof`.

But maybe I'm not getting it right or I'm missing something.

Reply:
the selection depends on the current environment, eg. linux vs mac. And this is not gonna change during execution. So the log message is printed only one time on node start. I'd say pretty much the same as it's done in `Secp256k1.initialize()`

Reply:
Got it now, thanks!